.github/workflows/publish-to-pypi.yml

```
@@ -0,0 +1,44 @@
+name: Release
+
+on:
+  release:
```

Why not test it per PR? You have it set on L11 that it runs for the PR if label called "Build wheels" is slapped on it. Of course, you have to manually create that label for this repo first.
```suggestion
on:
  pull_request:
  release:
```